test/db/SQLJSPackageDB.test.ts

```diff
@@ -0,0 +1,600 @@
+    });
+
+    it('should find resources that match a package name and package version', () => {
+      const resources = packageDb.findResourceInfos('*', {
+        scope: 'RegularPackage|3.2.2'
+      });
+      expect(resources).toHaveLength(2);
+      expect(resources).toContainEqual(expect.objectContaining(patientProfile));
+      expect(resources).toContainEqual(expect.objectContaining(valueSetThree));
+    });
+
+    it('should find resources that match one of the provided types', () => {
+      const resources = packageDb.findResourceInfos('*', {
+        type: ['Extension', 'ValueSet']
+      });
+      expect(resources).toHaveLength(3);
+      expect(resources).toContainEqual(expect.objectContaining(specialExtension));
+      expect(resources).toContainEqual(expect.objectContaining(valueSetThree));
+      expect(resources).toContainEqual(expect.objectContaining(valueSetFour));
+    });
+
+    it('should limit the number of results based on the limit option', () => {
+      const resources = packageDb.findResourceInfos('*', {
+        limit: 2
+      });
+      expect(resources).toHaveLength(2);
+    });
+  });
+
+  describe('#findResourceInfo', () => {
+    let packageDb: SQLJSPackageDB;
+    const specialExtension: ResourceInfo = {
+      resourceType: 'StructureDefinition',
+      id: 'a-special-extension',
+      name: 'SpecialExtension',
+      url: 'http://example.org/Extensions/a-special-extension',
+      sdFlavor: 'Extension',
+      packageName: 'RegularPackage',
+      packageVersion: '4.5.6'
+    };
+    const valueSetThree: ResourceInfo = {
+      resourceType: 'ValueSet',
+      id: 'my-value-set',
+      url: 'http://example.org/ValueSets/my-value-set',
+      name: 'MyValueSet',
+      packageName: 'RegularPackage',
+      packageVersion: '3.2.2'
+    };
+    const valueSetFour: ResourceInfo = {
+      resourceType: 'ValueSet',
+      id: 'my-value-set',
+      url: 'http://example.org/ValueSets/my-value-set',
+      name: 'MyValueSet',
+      packageName: 'RegularPackage',
+      packageVersion: '4.5.6'
+    };
+
+    beforeEach(() => {
+      packageDb = new SQLJSPackageDB(sqlDb);
+      packageDb.saveResourceInfo(specialExtension);
+      packageDb.saveResourceInfo(valueSetThree);
+      packageDb.saveResourceInfo(valueSetFour);
+    });
+
+    it('should return one resource when there is at least one match', () => {
+      const resource = packageDb.findResourceInfo('my-value-set');
+      expect(resource).toBeDefined();
+      // both valueSetThree and valueSetFour have a matching id,
+      // so either packageVersion is acceptable.
+      expect(resource).toEqual(
+        expect.objectContaining({
+          resourceType: 'ValueSet',
+          id: 'my-value-set',
+          url: 'http://example.org/ValueSets/my-value-set',
+          name: 'MyValueSet',
+          packageName: 'RegularPackage',
+          packageVersion: expect.stringMatching(/^(3\.2\.2|4\.5\.6)$/)
+        })
+      );
```

Comment:
Technically, I think this should always return `valueSetFour` (version `4.5.6`) because we order by `rowid DESC`. This is intentional to match current `FHIRDefinitions` behavior (which is "last in wins").